plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,818 @@
+        endtime_obj = datetime.strptime(end_time, TIME_FORMAT_WITHOUT_OFFSET)
+        if starttime_obj > endtime_obj:
+            self.module.exit_json(failed=True, msg=END_START_TIME.format(end_time, start_time))
+        return True
+
+    def check_diagnostics_jobs(self):
+        res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, MANAGERS_URI)
+        job_uri = f"{res_uri[0]}/{OEM}/{MANUFACTURER}/{JOBS}{JOBS_EXPAND}"
+        job_resp = self.idrac.invoke_request(job_uri, "GET")
+        job_list = job_resp.json_data.get('Members', [])
+        job_id = ""
+        for jb in job_list:
+            if jb.get("JobType") == "RemoteDiagnostics" and jb.get("JobState") in ["Scheduled", "Running", "Starting", "New"]:
+                job_id = jb['Id']
+                job_uri = f"{res_uri[0]}/{OEM}/{MANUFACTURER}/{JOBS}/{job_id}"
+                job_details = self.idrac.invoke_request(job_uri, "GET")
+                job_dict = remove_key(job_details.json_data, regex_pattern=ODATA_REGEX)
+                break
+        if self.module.check_mode and job_id:
+            self.module.exit_json(msg=NO_CHANGES_FOUND_MSG)
+        if self.module.check_mode and not job_id:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+        if job_id:
+            self.module.exit_json(msg=ALREADY_RUN_MSG, job_details=job_dict, skipped=True)
+
+
+class ExportDiagnostics:
+    STATUS_SUCCESS = [200, 202]
+
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+        self.export_url = None
+
+    def execute(self):
+        self.__get_export_diagnostics_url()
+        if self.module.check_mode:
+            self.perform_check_mode()
+        job_status = {}
+        self.__check_file_extension()
+        share_type = self.module.params.get('share_parameters').get('share_type')
+        share_type_methods = {
+            "local": self.__export_diagnostics_local,
+            "http": self.__export_diagnostics_http,
+            "https": self.__export_diagnostics_http,
+            "cifs": self.__export_diagnostics_cifs,
+            "nfs": self.__export_diagnostics_nfs
+        }
+        export_diagnostics_status = share_type_methods[share_type]()
+        if share_type_methods[share_type] != self.__export_diagnostics_local:
+            job_status = self.get_job_status(export_diagnostics_status)
+        status = export_diagnostics_status.status_code
+        if status in self.STATUS_SUCCESS:
+            msg = SUCCESS_EXPORT_MSG
+            job_details = job_status
+            return msg, job_details
+
+    def __export_diagnostics_local(self):
+        payload = {}
+        payload["ShareType"] = "Local"
+        path = self.module.params.get('share_parameters').get('share_name')
+        if not (os.path.exists(path)):
+            self.module.fail_json(msg=INVALID_DIRECTORY_MSG.format(path=path))
```

Comment:
lets use exit_json here to prevent traceback issue.

Reply:
done